clang/include/clang/Basic/Diagnostic.h

```diff
@@ -892,6 +894,8 @@ class DiagnosticsEngine : public RefCountedBase<DiagnosticsEngine> {
   /// \param FormatString A fixed diagnostic format string that will be hashed
   /// and mapped to a unique DiagID.
   template <unsigned N>
+  // TODO: Deprecate this once all uses are removed from LLVM
+  // [[deprecated("Use a CustomDiagDesc instead of a Level")]]
```

Comment:
```suggestion
  // TODO: Deprecate this once all uses are removed from Clang.
  // [[deprecated("Use a CustomDiagDesc instead of a Level")]]
```
I kind of wonder if we should have `-Wno-error=deprecated` in our CMake scripts and then just enable these sort of deprecations. We're terrible about ever actually removing uses of deprecated APIs and I think it's because they don't cause enough pain for us.